content/upcoming/language_migrations.md

```diff
@@ -0,0 +1,522 @@
+
+* One common point I found with Erik's results about **Go** is that it belongs to people who gave up studying **Rust**.
+
+## Over the last 16 years
+
+As we mentioned earlier before summing the transition matrices over time, we now consider specific years and determine how these yearly matrices look like. Do they express the same find of language profile ? How has it been evolving since the early 2000's ? Here is a sample of 4 matrices from various timeline intervals:
+
+<div class="grid2x">
+<div>
+<div>
+<img src="/post/language_migrations/P_2005.svg">
+<p align="center" class="dt"><pre><code class="hljs python">2005 - 2006</code></pre></p>
+</div>
+<div>
+<img src="/post/language_migrations/P_2011.svg">
+<p align="center" class="dt"><pre><code class="hljs python">2011 - 2012</code></pre></p>
+</div>
+</div>
+<div>
+<div>
+<img src="/post/language_migrations/P_2008.svg">
+<p align="center" class="dt"><pre><code class="hljs python">2008 - 2009</code></pre></p>
+</div>
+<div>
+<img src="/post/language_migrations/P_2015.svg">
+<p align="center" class="dt"><pre><code class="hljs python">2015 - 2016</code></pre></p>
+</div>
+</div>
+</div>
+
+<p align="center" class="caption">Temporal evolution of transition matrices</p>
+
+In the end, the evolution of these matrices over time seems to be affine, and we observe the same language profile every year. Thus, to highlight the time-line of this language profile, we apply the power iteration to each matrix. Earlier, we averaged the stationary distribution over the last 16 years but now we look at its chronological sequence. The resulting series of dominant eigenvectors is presented below on a stack area plot.
+
+<img src="/post/language_migrations/eigenvect_stack_22lang.png" class="fig">
+<p align="center" class="caption">Stationary distribution of languages over the last 16 years</p>
+
+The thickness of each band corresponds to the value in the dominant eigenvector. The bands are sorted by the averaged popularity which we calculated before.
+
+* The first two languages, **Python** and **Java** have the same profile. They have been taking the place of **C** for 15 years. Indeed the aggregation of these first 3 layers give a straight one.
+
+* The Attractiveness of **C++** dropped prominently in 2008 in favor of languages like **Ruby**. Nevertheless it has been sustaining its popularity ever since this period.
+
+* I definitely support Erik's conclusion saying **Perl** is dying.
+
+* For instance, if we consider the local event when **Apple** presented **Swift** on [WWDC'2014](https://developer.apple.com/videos/play/wwdc2014/402/), the layer of **Objective-C** which was its former main programming language should decrease for the benefit of this new alternative both safer and easier to use. Looking at the figure, this hypothesis turns out to be right.
```

Comment:
Bad. Rewrite the whole paragraph. Be simpler. Just write the facts. Apple presented Swift in ... and it was supposed to replace Obj-C. So Obj-C adoption should start to decrease after that event, but the sum of both languages should remain the same. Looking at the figure, this hypothesis turns out to be right.